src/transformers/file_utils.py

```diff
@@ -2112,6 +2112,112 @@ def _resumable_file_manager() -> "io.BufferedWriter":
     return cache_path
 
 
+def get_file_from_repo(
```

Comment:
Wondering a bit whether this function could directly go into `huggingface_hub` for other libraries to use as well

Reply:
This is a very similar method to the [`hf_hub_download`](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/file_download.py#L488) in `huggingface_hub`.

This method will eventually support the same errors as the `get_file_from_repo` method, and once it does it'd be super nice to leverage it from `huggingface_hub` indeed.